src/main/scala/com/mozilla/telemetry/sinks/HttpSink.scala

```diff
@@ -12,16 +15,77 @@ import scala.util.{Failure, Success, Try}
 
 object HttpSink {
 
+  /**
+    * Configuration options for HttpSink subclasses.
+    *
+    * These could all be constructor parameters, but we use a separate case class instead to
+    * avoid having to duplicate and keep the parameter list in sync across many subclasses.
+    */
   case class Config(
     maxAttempts: Int = 5,
     defaultDelayMillis: Int = 500,
     maxDelayMillis: Int = 30000,
     connectionTimeoutMillis: Int = 2000,
     readTimeoutMillis: Int = 5000,
     successCodes: Set[Int] = Set(OK),
-    retryCodes: Set[Int] = RetryCodes
-  )
+    retryCodes: Set[Int] = RetryCodes,
+    metrics: Option[Metrics] = None
+  ) {
+    /** Return a copy of this config with HttpSink metrics enabled. */
+    def withMetrics(implicit sc: SparkContext): Config = {
```

Comment:
I'd opt for exposing `SparkSession` in interfaces - we'll end up with a little less code when calling them.

Reply:
Seems reasonable. I'll use implicit SparkSession consistently in this chain.